### PR TITLE
Cleanup fixes

### PR DESCRIPTION
**Pull request by melchor629**
- fix: S3 cleanup was not working and even failed
- fix: `mdbackup cleanup` command did not load cloud providers from secret backends
- fix: documentation will only generate for the latest patch version in a minor series

### CODE REVIEW

File: docs/docker/versioning-build-grab-latest-minor.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        return str(self).__hash__()
+
+
+versions = [Version(line[:-1]) for line in sys.stdin]
+versions.sort()
+grouped_versions = [(version, Version(version.major, version.minor, 0)) for version in versions]
+grouped_versions = {key: [v for v, _ in values]
+                    for key, values in itertools.groupby(grouped_versions, key=lambda x: x[1])}
+final_versions = [next(iter(sorted(grouped_versions[key], reverse=True))) for key in sorted(grouped_versions.keys())]
+[print(str(final_version)) for final_version in final_versions]
```
Comment (melchor629) [inline docs/docker/versioning-build-grab-latest-minor.py:82]: This big script what it does is to
1. Parse versions to `Version` object (which contains versions split in several fields [major, minor, patch, extra] that matches `v<major>.<minor>.<patch>[-extra]` format)
2. Group versions by major+minor (that is `v0.1.0` and `v0.1.1` will be in the same group, but `v0.2.2` not)
3. Print only the more recent version of each group

File: mdbackup/__main__.py [hunk 2]
```diff
@@ -126,18 +126,16 @@ def main():
     _register_actions(config)
 
     try:
+        _load_secrets(config)
```
Comment (melchor629) [inline mdbackup/__main__.py:129]: Secrets must be loaded always, even for cleanup. This fixes one bug :)

**Review by amgxv — DISMISSED** [review]

File: mdbackup/storage/s3.py [hunk 3]
```diff
@@ -41,13 +41,12 @@ def list_directory_recursive(self, path: Union[str, Path]) -> List[str]:
                  for key in [item['Key'][len(self.__pre):].lstrip('/')
                              for item in res.get('Contents', [])]
                  if key != '']
-        self.__log.debug(items)
         return items
 
     def list_directory(self, path: Union[str, Path]) -> List[str]:
         items = self.list_directory_recursive(path)
         rep_path = path + '/' if path != '' else path
-        items = [key[len(self.__pre):].lstrip('/') for key in items
```
Comment (melchor629) [inline mdbackup/storage/s3.py:50]: This fixes the bug in which folders were created with wrong names

File: mdbackup/storage/s3.py [hunk 4]
```diff
@@ -72,27 +71,25 @@ def upload(self, path: Path, parent: Union[Path, str, str] = None):
...
+                              key,
+                              ExtraArgs={
+                                  'ContentType': magic.from_file(str(path.absolute()), mime=True),
+                                  'StorageClass': self.__storageclass,
+                              })
 
     def delete(self, path: Union[Path, str]):
-        path = self.__ok_key(path)
-        self.__log.info(f'Deleting {self.__pre}/{path}')
+        full_path = self.__ok_key(path)
```
Comment (melchor629) [inline mdbackup/storage/s3.py:83]: This fixes the delete method which created a wrong path and tried to remove it: `backups/pi/backups/pi/sth` now will be `backups/pi/sth`

File: mdbackup/storage/s3.py [hunk 5]
```diff
@@ -72,27 +71,25 @@ def upload(self, path: Path, parent: Union[Path, str, str] = None):
...
+        self.__log.info(f'Deleting {full_path}')
         objects_to_delete = self.list_directory_recursive(path)[::-1]
         while len(objects_to_delete) > 0:
-            ret = self.__s3.delete_objects(
+            self.__s3.delete_objects(
                 Bucket=self.__bucket,
                 Delete={
-                    'Objects': [{'Key': self.__ok_key(key)} for key in objects_to_delete],
+                    'Objects': [{'Key': f'{self.__pre}/{key}'} for key in objects_to_delete],
```
Comment (melchor629) [inline mdbackup/storage/s3.py:90]: The `__ok_key()` method removed the final `/` and folders were not removed. Now they will be removed properly :)

**Review by amgxv — APPROVED** [review]

**Pull request merged**